internal/impl/kafka/sasl.go

```diff
@@ -230,6 +236,26 @@ func SaramaSASLField() *service.ConfigField {
 		service.NewStringField(saramaFieldSASLTokenKey).
 			Description("Required when using a `token_cache`, the key to query the cache with for tokens.").
 			Default(""),
+		service.NewObjectField(saramaFieldSASLAws, []*service.ConfigField{
+			service.NewStringField("region").
+				Description("The AWS region to target.").
+				Default("").
+				Advanced(),
+			service.NewObjectField("credentials",
+				service.NewStringField("profile").
+					Description("A profile from `~/.aws/credentials` to use.").
+					Default("").Advanced(),
+				service.NewStringField("role").
+					Description("A role ARN to assume.").
+					Default("").Advanced(),
+				service.NewStringField("role_external_id").
+					Description("An external ID to provide when assuming a role.").
+					Default("").Advanced()).
+				Advanced().
+				Description("Optional manual configuration of AWS credentials to use. More information can be found [in this document](/docs/guides/cloud/aws)."),
```

Comment:
Perhaps a bit nit picky - but maybe here we remove the link to the /docs/guides/cloud/aws page - because is a different object on that page: 

```yaml
credentials:
  profile: ""
  id: ""
  secret: ""
  token: ""
  role: ""
  role_external_id: ""
```

And we list the ways & order in which we try to find credentials like this:

```suggestion
				Description(`Optional manual configuration of AWS credentials to use. An Auth token will be made from AWS credentials using the following config fields:
 - ` + "`profile`" +
					`
 - ` + "`role` & `role_external_id`" +
					`
 - ` + "`role`" +
					`
 - ` + "Or leaving these fields blank will load AWS Credentials according to AWS's [LoadDefaultConfig](https://docs.aws.amazon.com/sdk-for-go/v2/developer-guide/configure-gosdk.html#specifying-credentials)"),
```

Reply:
If you decide to alter  - don't forget to `make docs`